### PR TITLE
Derived validity

### PR DESCRIPTION
**Pull request by jkburges**
I acknowledge that this is a **huge** PR - sorry :-)  I will note what the high level changes are in this PR in order to make it a *little* easier to review.

1. Detection status is now derived on-the-fly, from associated receiver/deployment/recovery
1. No longer any need to `rescan` invalid detections when any of the above change
1. Tidied code along the way: code style, removed currently unused (and unlikely to return) KML code
1. Integration tests now run migrations at the beginning, rather than the DB being initialised by hibernate
1. Removed some unused controller code, e.g. edit/update of entities that should never be updated from the UI

### CODE REVIEW

**Comment by jkburges** [comment]
@dnahodil want to make a date for tomorrow AM to go through this?

**Comment by dnahodil** [comment]
:ok_hand:

**Comment by dnahodil** [comment]
The DetectionTests look all good. I think they test the cases we're looking at. I've just commented a few thoughts from looking in there but nothing too serious.

File: src/groovy/au/org/emii/aatams/detection/DetectionView.groovy [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    // GORM-like methods.
+    def refresh(dataSource) {
+        get(this.id, dataSource, this)
+    }
+
+    static Detection get(id, dataSource) {
+        return get(id, dataSource, [:])
+    }
+
+    static Detection get(id, dataSource, detection) {
```
Comment (dnahodil) [inline src/groovy/au/org/emii/aatams/detection/DetectionView.groovy]: If you wanted you could add a default value for detection thereby making the other get() method redundant.

```groovy
static Detection get(id, dataSource, detection = [:]) {
    ...
}
```

File: grails-app/migrations/load_all_dets.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# TODO: dos2unix
```
Comment (dnahodil) [inline grails-app/migrations/load_all_dets.sh]: I've added a ticket for this on the board, @jkburges so we don't forget

**Comment by dnahodil** [comment]
That's about all I have in the way of review comments, the work is good. Just a few things I commented on but they're not big deals.

**Comment by jkburges** [comment]
GTG again I think @dnahodil.  Your style of commenting on commit does make it a little difficult to confirm this, I must say (because one has to go and open up each commit in github, rather than just viewing in line here).

**Comment by dnahodil** [comment]
Yeah, sorry about that. Although I generally go through big ones commit-by-commit I try not to comment that way (even though I obviously did yesterday).

**Pull request merged**